modules/core/src/main/java/org/apache/ignite/internal/close/AutoCloser.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.close;
+
+/**
+ * A utility that allows to use something that is not an {@link AutoCloseable} in try-with-resources blocks.
+ *
+ * @param <T> Type of the resource.
+ * @param <X> Exception that can be thrown by the closer during closing. This type parameter is needed to make sure
+ *     that {@link AutoCloser#close()} throws declaration is as narrow as possible.
+ */
+@SuppressWarnings("CheckedExceptionClass")
+public class AutoCloser<T, X extends Exception> implements AutoCloseable {
```

Comment:
Well, to be honest all that stuff seems as an over design to me.

Reply:
Removed the stuff